libs/npm/core/src/components/grid-table/PagedTable.tsx

```diff
@@ -22,8 +22,15 @@ export const PagedTable = <CT extends object = Record<string, unknown>>({
   ...rest
 }: IPagedTableProps<CT>) => {
   const [items, setItems] = React.useState<CT[]>(page.items);
+  const [addToInfiniteItems, setAddToInfiniteItems] = React.useState<boolean>(false);
   React.useEffect(() => {
-    infiniteScroll && setItems(items.concat(page.items));
+    if (addToInfiniteItems) {
+      setItems((prevItems) => prevItems.concat(page.items));
+      setAddToInfiniteItems(false);
+    } else {
+      // we don't want stale items when user changes sorting or filter from something other than scrolling
+      setItems(page.items);
+    }
     // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Spent a couple hours on the below

Reply:
99% of the time it means we're doing something wrong :(  But sometimes I have to do it too lol

Reply:
Lol one day we will get it.

I found out one good thing from the process at least. Often times if we are doing something like 
`setState({ ...state, someAttribute: x })` this could lead to problems with the dependencies but we can access the previous state with `setState((previousState) => { ...previousState, someAttribute: x })`